core/src/main/java/org/elasticsearch/search/highlight/Highlighters.java

```diff
@@ -18,44 +18,74 @@
  */
 package org.elasticsearch.search.highlight;
 
-import com.google.common.collect.ImmutableMap;
-import org.elasticsearch.common.collect.MapBuilder;
 import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.logging.DeprecationLogger;
+import org.elasticsearch.common.logging.ESLoggerFactory;
 import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.util.ExtensionPoint;
 
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
 
 /**
  *
  */
-public class Highlighters {
+public class Highlighters extends ExtensionPoint {
```

Comment:
Maybe call this `HighlightersExtensionPoint`? Also the type could be moved here?
```
public class HighlightersExtensionPoint extends ExtensionPoint<Highlighter> {
```